.github/workflows/tf-account.yaml

```diff
@@ -90,6 +120,23 @@ jobs:
         if: steps.plan.outcome == 'failure'
         run: exit 1
 
+      - name: Commit Version Lockfile
+        id: commit-lockfile
+        if: env.commit_lock == 'true' && github.ref == env.ROOT_BRANCH && github.event_name == 'push'
```

Comment:
There's a slight issue here, in that when we might be getting different providers when we open the PR vs when we merge to main, since we're doing `terraform init -upgrade` at different moments in time.